packages/backend/src/tokens/authenticateContext.ts

```diff
@@ -38,45 +48,157 @@ class AuthenticateContext {
+  }
+
+  private initHandshakeValues() {
+    this.devBrowserToken =
+      this.getQueryParam(constants.QueryParameters.DevBrowser) ||
+      this.getSuffixedOrUnSuffixedCookie(constants.Cookies.DevBrowser);
+    // Using getCookie since we don't suffix the handshake token cookie
+    this.handshakeToken =
+      this.getQueryParam(constants.QueryParameters.Handshake) || this.getCookie(constants.Cookies.Handshake);
   }
 
   private stripAuthorizationHeader(authValue: string | undefined | null): string | undefined {
     return authValue?.replace('Bearer ', '');
   }
+
+  private getQueryParam(name: string) {
+    return this.clerkRequest.clerkUrl.searchParams.get(name);
+  }
+
+  private getHeader(name: string) {
+    return this.clerkRequest.headers.get(name) || undefined;
+  }
+
+  private getCookie(name: string) {
+    return this.clerkRequest.cookies.get(name) || undefined;
+  }
+
+  private getSuffixedCookie(name: string) {
+    return this.getCookie(`${name}_${this.cookieSuffix}`) || undefined;
+  }
+
+  private getSuffixedOrUnSuffixedCookie(cookieName: string) {
+    if (this.suffixedCookies) {
+      return this.getSuffixedCookie(cookieName);
+    }
+    return this.getCookie(cookieName);
+  }
+
+  private shouldUseSuffixed(): boolean {
+    const suffixedClientUat = this.getSuffixedCookie(constants.Cookies.ClientUat);
+    const clientUat = this.getCookie(constants.Cookies.ClientUat);
+    const suffixedSession = this.getSuffixedCookie(constants.Cookies.Session) || '';
+    const session = this.getCookie(constants.Cookies.Session) || '';
+
+    // If there is no suffixed cookies use un-suffixed
+    if (!suffixedClientUat && !suffixedSession) {
+      return false;
+    }
+
+    // If there's a token in un-suffixed, and it doesn't belong to this
+    // instance, then we must trust suffixed
+    if (session && !this.tokenBelongsToInstance(session)) {
+      return true;
+    }
+
+    const { data: sessionData } = decodeJwt(session);
+    const sessionIat = sessionData?.payload.iat || 0;
+    const { data: suffixedSessionData } = decodeJwt(suffixedSession);
+    const suffixedSessionIat = suffixedSessionData?.payload.iat || 0;
+
+    // Both indicate signed in, but un-suffixed is newer
+    // Trust un-suffixed because it's newer
+    if (suffixedClientUat !== '0' && clientUat !== '0' && sessionIat > suffixedSessionIat) {
+      return false;
+    }
+
+    // Suffixed indicates signed out, but un-suffixed indicates signed in
+    // Trust un-suffixed because it gets set with both new and old clerk.js,
+    // so we can assume it's newer
+    if (suffixedClientUat === '0' && clientUat !== '0') {
+      return false;
+    }
+
+    // In production, we can trust suffixed_uat because we don't need clerk.js to set it
+    if (this.instanceType === 'production') {
+      if (suffixedSession && suffixedClientUat === '0') {
+        return false;
```

Comment:
I don't understand why we're returning false here.

This scenario means the user is signed out. We need to make sure it's returning a signed out state, regardless of what's happening in the unsuffixed case.